test/src/materials/InterfaceValueMaterial.C

```diff
@@ -0,0 +1,93 @@
+  params.addRequiredParam<std::string>("mat_prop_slave",
+                                       "The material property on the slave side of the interface");
+  params.addRequiredParam<std::string>("mat_prop_out_basename",
+                                       "The base name for the output material property");
+  params.addRequiredCoupledVar(
+      "var_master",
+      "A variable on the master side of the interface that should be equivalent to the value of "
+      "the master material property (through MaterialRealAux for example");
+  params.addRequiredCoupledVar(
+      "var_slave",
+      "A variable on the slave side of the interface that should be equivalent to the value of "
+      "the slave material property (through MaterialRealAux for example");
+  params.addRequiredCoupledVar("nl_var_master",
+                               "Master side non-linear variable for jump computation");
+  params.addRequiredCoupledVar("nl_var_slave",
+                               "Slave side non-linear variable for jump computation");
+  params.addRequiredParam<std::string>("mat_prop_var_out_basename",
+                                       "The base name for the output material property");
+  params.addParam<MooseEnum>("interface_value_type",
+                             InterfaceValueTools::InterfaceAverageOptions(),
+                             "Type of scalar output");
+  return params;
+}
+
+InterfaceValueMaterial::InterfaceValueMaterial(const InputParameters & parameters)
+  : InterfaceMaterial(parameters),
+    _mp_master_name(getParam<std::string>("mat_prop_master")),
+    _mp_slave_name(getParam<std::string>("mat_prop_slave")),
+    _mp_master(getMaterialPropertyByName<Real>(_mp_master_name)),
+    _mp_slave(getNeighborMaterialProperty<Real>(_mp_slave_name)),
+    _var_master(coupledValue("var_master")),
+    _var_slave(coupledNeighborValue("var_slave")),
+    _nl_var_master(coupledValue("nl_var_master")),
+    _nl_var_slave(coupledNeighborValue("nl_var_slave")),
+
+    _interface_value_type(parameters.get<MooseEnum>("interface_value_type")),
+    _mp_out_base_name(getParam<std::string>("mat_prop_out_basename")),
+    _mp_var_out_base_name(getParam<std::string>("mat_prop_var_out_basename")),
+    _interface_value(
+        declareProperty<Real>(_mp_out_base_name + "_" + std::string(_interface_value_type))),
+    _interface_value_2(
+        declareProperty<Real>(_mp_var_out_base_name + "_" + std::string(_interface_value_type))),
+    _interface_value_old(
+        getMaterialPropertyOld<Real>(_mp_out_base_name + "_" + std::string(_interface_value_type))),
+    _interface_value_2_old(getMaterialPropertyOld<Real>(_mp_var_out_base_name + "_" +
+                                                        std::string(_interface_value_type))),
+    _jump(declareProperty<Real>(static_cast<std::string>(_interface_value_type) + "_jump"))
+{
+}
+
+void
+InterfaceValueMaterial::computeQpProperties()
+{
+  mooseAssert(_mp_master[_qp] == _var_master[_qp],
```

Comment:
Not obvious to me why we need these two assertions here.

Reply:
This was initially a `mooseWarning` created by @arovinelli; I changed it to a `mooseAssert`. He developed this test to make sure that he could feed in a non-zero neighbor var and a neighbor material property and make sure that they match (with something like `MaterialRealAux` involved somewhere)

Reply:
yep @lindsayad perfectly got the idea.
This was a check to make sure that using a material property or an aux variable would not affect the value of the jump. I agree with modifying it in an assert